Can't find pybind11

# Bug reports:

I have pybind11 installed in `/usr/local/include/pybind11` and now it can't be found. 
```
CMake Error at cmake/SetupPybind11.cmake:22 (find_package):
  Could not find a package configuration file provided by "pybind11"
  (requested version 2.6.0) with any of the following names:

    pybind11Config.cmake
    pybind11-config.cmake

  Add the installation prefix of "pybind11" to CMAKE_PREFIX_PATH or set
  "pybind11_DIR" to a directory containing one of the above files.  If
  "pybind11" provides a separate development package or SDK, be sure it has
  been installed.
Call Stack (most recent call first):
  CMakeLists.txt:95 (include)
```
This is a regression in #2672 as far as I can tell. We can't rely on any CMake modules being provided. This will be a complete disaster on supercomputers. All we should need is the include directory (this is the whole benefit of header-only libs)

--- a/cmake/SetupPybind11.cmake
+++ b/cmake/SetupPybind11.cmake
@@ -19,7 +19,14 @@ if(BUILD_PYTHON_BINDINGS)
       "determined CMake dir: ${PYBIND11_CMAKEDIR}")
   endif()
 
-  find_package(pybind11 2.6.0 REQUIRED HINTS "${PYBIND11_CMAKEDIR}")
+  find_package(pybind11 2.6.0 REQUIRED
+    HINTS
+    ${PYBIND11_CMAKEDIR}
+    ${Python_SITEARCH}
+    ${Python_SITELIB}
+    ${Python_STDARCH}
+    ${Python_STDLIB}
+    )
 
   set_property(
     GLOBAL APPEND PROPERTY SPECTRE_THIRD_PARTY_LIBS

